From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 31 Mar 2025 17:01:48 -0500
Subject: [PATCH] flang: Add install targets for mod/header

Create targets to include ISO_Fortran_binding.h and the .mod files since they don't have ones.

Origin: vendor
Forwarded: not-needed
Last-Update: 2025-03-31
---
 flang/CMakeLists.txt           | 5 ++++-
 flang/tools/f18/CMakeLists.txt | 6 ++++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/flang/CMakeLists.txt b/flang/CMakeLists.txt
index 070c39eb6e9a..641c8b1b7b45 100644
--- a/flang/CMakeLists.txt
+++ b/flang/CMakeLists.txt
@@ -533,9 +533,12 @@ configure_file(
   ${FLANG_SOURCE_DIR}/include/flang/ISO_Fortran_binding.h
   ${HEADER_BINARY_DIR}/ISO_Fortran_binding.h)
 
+add_llvm_install_targets(install-flang-binding
+  COMPONENT flang-binding)
 # And also install it into the install area
 get_clang_resource_dir(HEADER_INSTALL_DIR SUBDIR include)
 install(
   FILES include/flang/ISO_Fortran_binding.h
-  DESTINATION ${HEADER_INSTALL_DIR} )
+  DESTINATION ${HEADER_INSTALL_DIR} 
+  COMPONENT flang-binding)
 
diff --git a/flang/tools/f18/CMakeLists.txt b/flang/tools/f18/CMakeLists.txt
index cec4e2d81072..295a0538dc78 100644
--- a/flang/tools/f18/CMakeLists.txt
+++ b/flang/tools/f18/CMakeLists.txt
@@ -40,6 +40,8 @@ set(module_objects "")
 # cross compiled, and thus can't be executed on the build system and thus
 # can't be used for generating module files.
 if (NOT CMAKE_CROSSCOMPILING)
+add_llvm_install_targets(install-flang-mod
+  COMPONENT flang-mod)
   foreach(filename ${MODULES})
     set(depends "")
     set(opts "")
@@ -97,7 +99,7 @@ if (NOT CMAKE_CROSSCOMPILING)
       DEPENDS flang-new ${FLANG_SOURCE_DIR}/module/${filename}.f90 ${FLANG_SOURCE_DIR}/module/__fortran_builtins.f90 ${depends}
     )
     list(APPEND MODULE_FILES ${base}.mod)
-    install(FILES ${base}.mod DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/flang")
+    install(FILES ${base}.mod DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/flang" COMPONENT flang-mod)
 
     # If a module has been compiled into an object file, add the file to
     # the link line for the FortranRuntime library.
@@ -130,7 +132,7 @@ if (NOT CMAKE_CROSSCOMPILING)
         DEPENDS ${base}.mod
         COMMAND ${CMAKE_COMMAND} -E copy ${base}_kinds.mod ${base}_kinds.f18.mod)
     list(APPEND MODULE_FILES ${base}.mod ${base}.f18.mod ${base}_kinds.mod ${base}_kinds.f18.mod)
-    install(FILES ${base}.mod ${base}.f18.mod ${base}_kinds.mod ${base}_kinds.f18.mod DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/flang")
+    install(FILES ${base}.mod ${base}.f18.mod ${base}_kinds.mod ${base}_kinds.f18.mod DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/flang" COMPONENT flang-mod)
   else()
     message(STATUS "Not building omp_lib.mod, no OpenMP runtime in LLVM_ENABLED_PROJECTS")
   endif()
